doc/model.rst

```diff
@@ -0,0 +1,683 @@
+following arguments are offered:
+
+- name (str): key to store script under
+- script (str, optional): TorchScript code (only supported for non-colocated orchestrators)
+- script_path (str, optional): path to TorchScript code
+- device (str, optional): device for script execution, defaults to “CPU”
+- devices_per_node (int): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- first_device (int): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+
+You might use TorchScript scripts to represent individual models within the model.
+
+Example: Loading an TorchScript to the Model
+--------------------------------------------
+In this example, we demonstrate how to instruct SmartSim to load
+an TorchScript into the database at model runtime.
+
+**Python Script: Define a TorchScript for Model Purposes**
+
+Define the TorchScript code to a variable in the Python driver script:
+
+.. code-block:: python
+
+    torch_script_str = "def negate(x):\n\treturn torch.neg(x)\n"
+
+**SmartSim Model Integration:**
+
+Assuming an initialized ``Model`` named `smartsim_model`, we specify the
+following parameters to the ``Model.add_script()`` function:
+
+.. code-block:: python
+
+    smartsim_model.add_script(name="example_script", script=torch_script_str, device="GPU", devices_per_node=2, first_device=0)
+
+In this integration, we provide the following details:
+
+-  `name`: "example_script" - key to store script under
+-  `script`: torch_script_str - TorchScript code
+-  `device`: "CPU" - device for script execution
+-  `devices_per_node`: 2 - Use two GPUs per node.
+-  `first_device`: 0 - Start with 0 index GPU.
+
+When the model is started via ``Experiment.start()``, the TorchScript will be loaded to the
+orchestrator that is launched prior to the start of the model.
+
+=========================
+Data Collision Prevention
+=========================
+--------
+Overview
+--------
+SmartSim and SmartRedis can avoid key collision by prepending program-unique
+prefixes to Model workloads launched through SmartSim. For example, if you were
+to have two applications feeding data to a single database, who produced keys
+of the same name, upon requesting this information there would be a key collision
+since there is no yet uniqueness between the same tensor names. By enabling key
+prefixing on a Model, SmartSim will append the model name to each key produced
+by the application and sent to the database as such: `model_name.tensor_name`.
+
+This is done simply through functions offered by the Model API:
+
+* Model.register_incoming_entity(incoming_entity)
+* Model.enable_key_prefixing()
+* Model.disable_key_prefixing()
+* Model.query_key_prefixing()
+
+-------
+Example
+-------
+We provide a producer/consumer example that demonstrates
```

Comment:
Consider making a more focused summary intro, e.g. "This example demonstrates mitigation of key collisions when multiple models write to the same key name. The consumer is able to retrieve both values successfully by enabling key prefixing."

Reply:
I agree on this.  I think a little more motivation and context for the example would be good.